src/components/common/Dialogs/CreatePaymentDialog/TokenAmountInput/TokenAmountInput.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useMemo } from 'react';
+import { defineMessages } from 'react-intl';
+import { AddressZero } from '@ethersproject/constants';
+import { useFormContext } from 'react-hook-form';
+
+import { HookFormInput as Input, TokenSymbolSelector } from '~shared/Fields';
+import EthUsd from '~shared/EthUsd';
+import { getTokenDecimalsWithFallback } from '~utils/tokens';
+import { notNull } from '~utils/arrays';
+import { Colony } from '~types';
+
+// import NetworkFee from '../NetworkFee';
+
+import styles from './TokenAmountInput.css';
+
+const displayName =
+  'common.CreatePaymentDialog.CreatePaymentDialogForm.TokenAmountInput';
+
+const MSG = defineMessages({
+  amount: {
+    id: `${displayName}.amount`,
+    defaultMessage: 'Amount',
+  },
+  fee: {
+    id: `${displayName}.fee`,
+    defaultMessage: 'Network fee: {fee} {symbol}',
+  },
+  token: {
+    id: `${displayName}.address`,
+    defaultMessage: 'Token',
+  },
+});
+
+interface Props {
+  colony: Colony | undefined;
+  disabled: boolean;
+}
+
+const TokenAmountInput = ({ colony, disabled }: Props) => {
+  const { getValues } = useFormContext();
+  const values = getValues();
+  const colonyTokens =
+    colony?.tokens?.items
+      .filter(notNull)
+      .map((colonyToken) => colonyToken.token) || [];
+  const selectedToken = colonyTokens.find(
+    (token) => token?.tokenAddress === values.tokenAddress,
+  );
+  const formattingOptions = useMemo(
+    () => ({
+      delimiter: ',',
+      numeral: true,
+      numeralPositiveOnly: true,
+      numeralDecimalScale: getTokenDecimalsWithFallback(
+        selectedToken && selectedToken.decimals,
+      ),
+    }),
+    [selectedToken],
+  );
+
+  return (
+    <div className={styles.tokenAmount}>
+      <div className={styles.tokenAmountInputContainer}>
+        <Input
+          label={MSG.amount}
+          name="amount"
+          appearance={{
+            theme: 'minimal',
+            align: 'right',
+          }}
+          formattingOptions={formattingOptions}
+          disabled={disabled}
+          dataTest="paymentAmountInput"
+        />
+        {/* <NetworkFee colony={colony} networkFeeInverse={networkFeeInverse} customAmountError={customAmountError} /> */}
```

Comment:
Why is this commented out? It was showing up earlier

Reply:
Are you certain? This is a new component I introduced in the latest commits after your first review, and it should contain the code that shows the network fee. The code inside this component was also commented out before I extracted it in this component.

Why has this been commented out before and now: I think there's currently no query to get the network fee (In the Dapp you could get it like this: `const { feeInverse: networkFeeInverse } = useNetworkContracts();`).

Reply:
It could have been in the Dapp... That would make sense...